### PR TITLE
Line Graph Fix

### PR DESCRIPTION
**Pull request by rishab1065**
- Add xlabel & ylabel offset 
- Add xAxisLabel 
- Remove right side space with some data adjustments 
- Add hidePointsAtIndex prop

### CODE REVIEW

**Comment by rishab1065** [comment]
@Hermanya Please review and merge the PR

**Comment by rishab1065** [comment]
@Hermanya Please review and merge the PR

**Comment by rishab1065** [comment]
@Hermanya Please review and merge this PR

**Comment by Hermanya** [comment]
It's on my to-do list

File: src/line-chart.js [hunk 1]
```diff
@@ -209,12 +237,17 @@ class LineChart extends AbstractChart {
 
   renderBezierShadow = config => {
     const { width, height, paddingRight, paddingTop, data } = config;
+
     return data.map((dataset, index) => {
+      const dataSetLength =
+        dataset.data.length === 1
+          ? dataset.data.length - 1
```
Comment (Hermanya) [inline src/line-chart.js]: why not change this to 0? if we already know that `dataset.data.length` is `1`.

File: src/line-chart.js [hunk 2]
```diff
@@ -37,18 +37,31 @@ class LineChart extends AbstractChart {
...
     const output = [];
-    const datas = this.getDatas(data);
+    let datas = this.getDatas(data);
+
     const baseHeight = this.calcBaseHeight(datas, height);
     const getDotColor = this.props.getDotColor || null;
+    const hidePointsAtIndex = this.props.hidePointsAtIndex || [];
+    const {
+      propsForDots: { strokeWidth = 0, r = 0 } = {}
+    } = this.props.chartConfig;
```
Comment (Hermanya) [inline src/line-chart.js]: LineChart already has a `getPropsForDots` method.

File: src/abstract-chart.js [hunk 3]
```diff
@@ -99,15 +99,14 @@ class AbstractChart extends Component {
...
-      yLabelsOffset = 12,
       horizontalLabelRotation = 0
     } = config;
     const decimalPlaces =
       this.props.chartConfig.decimalPlaces === undefined
         ? 2
         : this.props.chartConfig.decimalPlaces;
     const yAxisLabel = this.props.yAxisLabel || "";
-
+    const yLabelsOffset = this.props.yLabelsOffset || 12;
```
Comment (Hermanya) [inline src/abstract-chart.js]: why did you move `yLabelsOffset` from the `chartConfig` into its own prop? that's a potentially breaking api change.

File: src/abstract-chart.js [hunk 4]
```diff
@@ -157,14 +156,19 @@ class AbstractChart extends Component {
     if (stackedBar) {
       fac = 0.71;
     }
-
+    const xAxisLabel = this.props.xAxisLabel || "";
+    const xLabelsOffset = this.props.xLabelsOffset || 0;
+    const hidePointsAtIndex = this.props.hidePointsAtIndex || [];
```
Comment (Hermanya) [inline src/abstract-chart.js]: in what use case do you use `hidePointsAtIndex`?

File: src/line-chart.js [hunk 5]
```diff
@@ -106,8 +121,13 @@ class LineChart extends AbstractChart {
             dataset.data
               .map((d, i) => {
                 const x =
-                  paddingRight +
-                  (i * (width - paddingRight)) / dataset.data.length;
+                  i === dataset.data.length - 1 && dataset.data.length !== 1
```
Comment (Hermanya) [inline src/line-chart.js]: this line repeats multiple times in this PR, why is it necessary? you could take it out into its own method to re-use it.

**Review by Hermanya — CHANGES_REQUESTED** [review]
The example app looks pretty broken right now
image
What issue were you trying to fix?

File: src/abstract-chart.js (hunk 3 above)
Comment (rishab1065) [reply src/abstract-chart.js]: AFAIK `yLabelsOffset` is not being used in the code from chart config. `renderHorizontalLabels` function accepts the config prop which has the value 
`{
      width,
      height,
      verticalLabelRotation,
      horizontalLabelRotation
    }` in `line-chart.js line no 277`.

**Comment by rishab1065** [comment]
@Hermanya I have updated the PR with the changes. Please review and let me know.

File: src/abstract-chart.js (hunk 3 above)
Comment (Hermanya) [reply src/abstract-chart.js]: it's used on line 123 of abstract chart

Comment (rishab1065) [reply src/abstract-chart.js]: @Hermanya I have moved it inside `chartConfig`.

**Comment by Hermanya** [comment]
This is now available in v3.11.0, thank you for your contribution 😺

**Comment by brandedgames** [comment]
The empty space on the right of the chart is still there on v6.6.1.

**Comment by hapo-nghialuu** [comment]
image
It's still happening for me.
